kotlin/plugin_crasher/src/main/kotlin/Main.kt

```diff
@@ -17,15 +17,31 @@ fun main(args: Array<String>) {
   when (args[0]) {
     "anr" -> injectAnr()
     "art-oom" -> injectArtOom()
+    "background-error" -> injectBackgroundThreadError()
     else -> usage(1)
   }
 }
 
 fun usage(exitCode: Int) {
-  eprintln("stoic crasher [anr|art-oom]")
+  eprintln("stoic crasher [anr|art-oom|background-error]")
   exit(exitCode)
 }
 
+/**
+ * This creates a HandlerThread named "Background Thread Error".
+ * Post a message to the HandlerThread that will throw an exception.
+ * This will cause the thread to crash.
+ */
+fun injectBackgroundThreadError() {
+  eprintln("Injecting background thread error...")
+  HandlerThread("Injected Background Thread Error").also {
```

Comment:
Can you use https://github.com/square/stoic/blob/main/kotlin/stoicAndroid/src/main/kotlin/com/square/stoic/Stoic.kt#L157C7-L157C18 for this?

I think something like this will work:

```
HandlerThread("injected-error").also {
  stoic.runOnLooper(it.looper, Integer.MAX_VALUE) {
    eprintln("Injecting background thread error...")
    throw Exception("This is a background thread error.")
  }
}
```

It's better to have short thread names without spaces to avoid problems with things that parse https://man7.org/linux/man-pages/man5/proc_pid_stat.5.html output.

The `runOnLooper` API just makes it so anything you print can still go to stdout/stderr.